### PR TITLE
Add documentation for creating a fork

### PR DESCRIPTION
**Pull request by joshleecreates**
Towards https://github.com/open-telemetry/opentelemetry-demo/issues/267#issuecomment-1276551813.

## Changes

Adds some documentation around environment variables and values overrides for creating forks and demos.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,6 +163,8 @@ In case you need to run a specific suite of tests you can execute
 - [Lightstep](https://github.com/lightstep/opentelemetry-demo)
 - [New Relic](https://github.com/newrelic/opentelemetry-demo)
 
+See [Forking](./docs/forking.md) for more information about creating your own fork.
```
Comment (reyang) [inline README.md:166]: Minor, maybe add some words here to encourage folks to add their forks here.

**Review by reyang — APPROVED** [review]
LGTM.

File: README.md (hunk 1 above)
Comment (joshleecreates) [reply README.md:166]: Added here: https://github.com/open-telemetry/opentelemetry-demo/pull/447/files#diff-1c2caaad726b38f555fa598c8c26fbec6f8a81c8793ce0074a280bf8f375d744R9

**Comment by cartersocha** [comment]
@joshleecreates could you make the helm command a shell code block? It looks a bit odd with regular text alongside it

```shell 

```

**Comment by cartersocha** [comment]
Lol my comment obscures the formatting but something like " ```shell "

Should work

**Comment by joshleecreates** [comment]
> Lol my comment obscures the formatting but something like " ```shell "
> 
> Should work

Yes will do.. lol.

**Comment by cartersocha** [comment]
Lgtm

**Review by cartersocha — APPROVED** [review]

**Pull request merged**